### PR TITLE
Psfex recipe

### PR DESCRIPTION
**Pull request by teake**
@conda-forge/help-c-cpp

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

__Note__: this package's name clashes with that of https://github.com/conda-forge/psfex-feedstock. I'm not sure what the best way to proceed is. The existing psfex conda forge package looks like a misnomer, [as it appears to be a utility to reconstruct images from psfex models](https://github.com/esheldon/psfex#psfex) (i.e. from files generated by the program in this PR, [psfex](https://psfex.readthedocs.io/en/latest/Introduction.html)).

If the existing conda forge package would have had zero downloads I'd vote to rename it (to `psfex-rec`, like its cli tool). But since that is not the case, I'm not sure how to proceed. I could rename the package name in the PR to `astromatic-psfex`, since it is part of the so-called [AstrOmatic software suite](https://www.astromatic.net/software). But I have more recipes for AstrOmatic utilities lined up, and this is the only one with a name clash. So I'm a bit hesitant to prefix all of them with `astromatic-`.

CC'ing the existing psf-feedstock maintainer @beckermr and the author of the underlying code @esheldon for input.

### CODE REVIEW

**Comment by beckermr** [comment]
@esheldon and I discussed this when we made the other recipe. I agree the namespace collision is unfortunate. I am happy to rename the python tool but I think we need to all agree on something before making a decision.

**Comment by teake** [comment]
Ok, @esheldon could you weigh in on this? And do we need to ping someone else?

**Comment by esheldon** [comment]
My view is that python utilities should have precedence in name space collisions.  There is only one native python psfex package, as far as I'm aware (there are some codes that just call out to the command line tool but that's different).  It is true that it should have been named differently from the beginning perhaps, since it is only a reader, but I also never intended anyone else to use it;  c'est la vie

I still think of the anaconda environment as primarily a place for python packages, and anything else is support for those packages.  From that point of view, it makes sense to me that non-python command line tools should be prefixed, e.g. with astromatic or whatever to make sure they won't collide with python packages.

However, I don't feel very strongly about all this in this particular case.  psfex is soon to be replaced by piff for all the work we do

**Comment by scopatz** [comment]
> I still think of the anaconda environment as primarily a place for python packages

I only agree with this insofar as Python is the most popular.  I have built out a lot of stacks on conda-forge that have nothing to do with Python, which is the primary benefit of using conda-forge.

I think it breaks a lot of reproducability and bot features if we start renaming feedstocks in this way.  So unfortunately, the namespace is first come, first served.

File: recipes/psfex/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+{% set name = "psfex" %}
+{% set version = "3.21.1" %}
+
+package:
+  name: {{ name }}
```
Comment (scopatz) [inline recipes/psfex/meta.yaml]: maybe name this `{{ name }}-c`

Comment (beckermr) [reply recipes/psfex/meta.yaml]: The astromatic- prefix makes more sense if we are not going to change things. This way we can include sextractor, swarp etc in the same way for consistency.

Comment (beckermr) [reply recipes/psfex/meta.yaml]: The other thing is that the dark energy survey has custom versions of these packages that I myself have been meaning to add using the des- prefix. At least this way everything would be consistent.

Comment (teake) [reply recipes/psfex/meta.yaml]: Going slightly off topic here, but if DES requires custom versions they should perhaps not go into `conda-forge` but a separate `des` channel? 

I'm pushing this recipe as part of an effort to migrate not only our own Astro-WISE software but also the Euclid SGS processing environment over to conda environments. The idea is to use as vanilla 3rd party packages as much as possible. If we need modifications, I will try to get them in upstream. Failing that, we will put modified packages in our own channel for Astro-WISE (I'm not sure what the Euclid policy will be as I'm only remotely involved). 

Anyway, I will change this to `astromatic-{{ name }}`.

**Comment by teake** [comment]
> I think it breaks a lot of reproducability and bot features if we start renaming feedstocks in this way. So unfortunately, the namespace is first come, first served.

Not what I wanted to hear, but I guess what I needed to hear. I do wish there was a package name vetting process of some sort, as the existing `psfex` conda forge package is a misnomer IMHO.

Anyway, I've renamed this to `astromatic-psfex` and will also prefix the other AstrOmatic packages similarly.

Thanks everybody for the input!

**Comment by scopatz** [comment]
for what it is worth, I think in general, you don't need to prefix with astromatic, just in cases like this

**Comment by scopatz** [comment]
also, there have been previous attempts to add namespaces to conda, it might be worth reviving those discussions.

 Thanks!

**Pull request merged**